include/boost/beast/core/detail/any_buffers.hpp

```diff
@@ -24,6 +24,34 @@ namespace boost {
 namespace beast {
 namespace detail {
 
+template<class ValueType>
```

Comment:
I would just use `bool IsConst` instead of `ValueType`, because `ValueType` is unconstrained while `bool` accurately reflects the actual cardinality of the value.